### DISCUSSION TITLE
@vercel/toolbar plugin not working with a next config in esm mode

### DISCUSSION THREAD

**Discussion by hrougier** [vercel/community#4776] [Bug Report]
### Page to Investigate

https://vercel.com/docs/workflow-collaboration/comments/in-production-and-localhost

### Steps to Reproduce

Having a `next.config.js` using the ESM syntax instead of CJS:

```js
import { withVercelToolbar } from '@vercel/toolbar/plugins/next'

export default withVercelToolbar()({
  // config...
})
```

Starting the dev server results in the following error:

```
[vercel-toolbar] Error running the get-free-port script ReferenceError: __dirname is not defined
    at L (file:///path/to/node_modules/@vercel/toolbar/dist/plugins/next.js:4:273)
    ...
```

Because of the `next.js` plugin file using the `__dirname` variable not available in ESM mode.

Would it be possible to make it compatible with ESM?

**Comment by o-az** [comment]
same issue here. I fixed it by manually edition `node_modules/@vercel/toolbar/dist/plugins/next.js` and adding `__dirname` at the top:

```js
import url from 'node:url'
import path from 'node:path'

const __dirname = path.dirname(url.fileURLToPath(import.meta.url))
```

**Comment by o-az** [comment]
Is the source code for this plugin public? I will happily contribute to fix this bug

**Comment by leerob** [comment]
This is fixed and released in version `0.1.9` 🙏

**Comment by amlcodes** [comment]


**Reply by amlcodes** [reply]
production only
